internal/archive/extract.go

```diff
@@ -40,29 +39,29 @@ type fileDownloader interface {
 	Get(string) (*http.Response, error)
 }
 
-type Extractor func(logr.Logger, context.Context, string, time.Duration) (*Extraction, error)
+type Extractor func(logr.Logger, context.Context, string, string, time.Duration) (*Extraction, error)
 
 type Extraction struct {
-	RootDir     string
 	Archive     string
 	ContentsDir string
 }
 
-func FetchAndExtract(log logr.Logger, ctx context.Context, url string, timeout time.Duration) (*Extraction, error) {
+func FetchAndExtract(log logr.Logger, ctx context.Context, url string, wd string, timeout time.Duration) (*Extraction, error) {
 	if timeout > 0 {
 		var cancel context.CancelFunc
 		ctx, cancel = context.WithTimeout(ctx, timeout)
 		defer cancel()
 	}
 
-	wd, err := ioutil.TempDir("", "forge-")
-	if err != nil {
+	if isDir, err := isDirectory(wd); err != nil {
 		return nil, err
+	} else if !isDir {
+		return nil, fmt.Errorf("%v is not a directory", wd)
```

Comment:
It seems like we really want to check for an error here and propagate it rather than evaluate a boolean.

```suggestion
if err := AssertDir(wd); err != nil {
      // we "wrap" the original error to give some more context
      return nil, fmt.Errorf("invalid build contex directory: %w", err)
```

where `AssertDir` looks like this:

```go
// exported func that we can unit test separately and/or move into a util pkg if we want
func AssertDir(path string) error {
	fi, err := os.Stat(path)
	if err != nil {
		return err
	}
	if !fi.IsDir() {
		return fmt.Errorf("%q is not a directory", path)
	}

	return nil
}

```

Reply:
Good idea. Done.